test/functional/rpc_generateblock.py

```diff
@@ -5,33 +5,70 @@
 '''Test generateblock rpc.
 '''
 
+from decimal import Decimal
 from test_framework.test_framework import BitcoinTestFramework
+from test_framework.address import ADDRESS_BCRT1_P2WSH_OP_TRUE
 from test_framework.util import (
     assert_equal,
     assert_raises_rpc_error,
 )
+from test_framework.messages import (
+    COIN,
+    COutPoint,
+    CTransaction,
+    CTxIn,
+    CTxInWitness,
+    CTxOut,
+)
+from test_framework.script import (
+    CScript,
+    OP_TRUE,
+)
+from test_framework.util import (
+    satoshi_round,
+    hex_str_to_bytes,
+)
+from test_framework.wallet import MiniWallet
+
+def create_miniwallet_rawtx(utxo, scriptPubKey):
```

Comment:
I think @MarcoFalke is setting you up for a layup here https://github.com/bitcoin/bitcoin/pull/20876. Then @glozow suggests "was just thinking that it'd be nice to have a just-create-don't-send option for the MiniWallet. What do you think of a bool option?"

I think this is the PR where this is needed. So I'd consider rebasing on-top of #20876, and adding this option to `send_self_transfer` as described here. Then you can utilize this extended functionality and ditch `create_miniwallet_rawtx`

thoughts?

Reply:
Yes I didn't like my approach but I wasn't sure I should touch the miniwallet class itself to add a `create_raw_transaction` function. So I think your suggestion is a nice solution.

Reply:
I did a similar thing in `mempool_limit` by adding a `prepare_tx` method in MiniWallet Class. 
link: https://github.com/bitcoin/bitcoin/pull/20874/files#diff-7932a60a9127fd22d10d367526fd7b987f9647ce017595f8b1af5c32d5db0083R58
To be honest, it was just to avoid repetition, but I can see how it could be helpful for both of us

Reply:
I did a similar thing in mempool_limit by adding a prepare_tx method in MiniWallet Class.
link: https://github.com/bitcoin/bitcoin/pull/20874/files#diff-7932a60a9127fd22d10d367526fd7b987f9647ce017595f8b1af5c32d5db0083R58
To be honest, it was just to avoid repetition, but I can see how it could be helpful for both of us

Reply:
nit: shouldn't the name of the method be `create_and_sign_miniwallet_rawtx`?

Reply:
Agree with @mjdietzx . Please rebase and add the bool option to #20876